app/models/services.py

```diff
@@ -31,13 +28,23 @@ def insert(shop_id="", name="", description="", images=None):
             "name": name,
             "slug": slugify(name),
             "description": description,
-            "images": images,
+            "price": price,
+            "images": images or [],
         }
     )
+    return id
 
-    # TODO: upload images into disk
-    pass
 
+def update(id="", shop_id="", name="", description="", price="", images=None):
+    query = tinydb.Query()
+    doc = {
+        "name": name,
+        "slug": slugify(name),
+        "description": description,
+        "price": price,
+        "images": images,
+    }
```

Comment:
We have few issues with this method:
1. the shop_id is not used?
2. we are not checking or validating the required fields, for example the `id` field is required here
3. in order for this method to work in its current implementation, we would need to always provide all data regardless if they are updated or not, I prefer to only pass the data we need to update, for example if I need to update only the name, then no need to pass the images. To fix it, we will need to build the doc based on the provided args

for example: 
```
doc = {}
if name:
     doc['name'] = name

if description:
    doc['description] = description

....
```

Reply:
1. the shop_id is not used? >>> Nice catch. I'll fix that
2. The id is always included in the request because it's a hidden field on the edit form, so I think only need to remove default value of id in update function. What do you think?
3. This is a good idea, I'll do it.

Reply:
sounds good! thank you!